src/components/rockets.js

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  getRockets,
+  reserveRocket,
+  cancelReservation,
+}
+  from '../redux/rockets/rocketsReducer';
+import styles from './rockets.module.css';
+
+const Rockets = () => {
+  const dispatch = useDispatch();
+  const rocketsList = useSelector((state) => state.rockets);
+  const reserveRocketHandler = (id) => {
+    dispatch(reserveRocket(id));
+  };
+
+  const cancelReservationHandler = (id) => {
+    dispatch(cancelReservation(id));
+  };
+
+  useEffect(() => {
+    if (rocketsList.length === 0) {
+      dispatch(getRockets());
+    }
+  }, []);
+
+  let reserved = null;
+
+  return (
+    <div id={styles.container}>
+      {rocketsList.map((rocket) => (
+        <ul key={rocket.id}>
+          <li>
+            <img className={styles.images} alt="rocket img" src={rocket.flickr_images} />
+          </li>
+          <div className={styles.info}>
+            <li>{rocket.name}</li>
+
+            <li
+              className={styles.description}
+            >
+              {rocket.reserved && (<span className={styles.reserved}>Reserved</span>)}
+              {rocket.description}
+            </li>
+
+            <button
+              className={!rocket.reserved === true ? styles.button : styles.cancel}
+              type="button"
+              onClick={() => {
+                reserved = rocket.reserved;
+                if (!reserved) {
+                  reserveRocketHandler(rocket.id);
+                } else {
+                  cancelReservationHandler(rocket.id);
+                }
+              }}
+            >
+              Reserve Rocket
```

Comment:
- [ ] In this button, please change the text of it when it's on the cancelation process to follow the given design and also to make more sense 😄 🙏 You could've done it as the missions' buttons or you could write a very simple js function that changes the text of the button 💯 
_Please check the image in the project requirements right [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#:~:text=My%20Profile%20section.-,Rockets%20/Dragons,-The%20Rockets%20section)._